plugins/Feedback/vue/src/FeedbackQuestion/FeedbackQuestion.vue

```diff
@@ -0,0 +1,173 @@
+          <div class="messageContainer">
+            <div class="error-text" v-if="errorMessage">{{ errorMessage }}</div>
+            <textarea id="message" :class="{'has-error':errorMessage}" v-model="feedbackMessage"/>
+          </div>
+          <br/>
+          <p
+            v-html="translate('Feedback_Policy',`<a rel='nofollow' href='https://matomo.org/privacy-policy/' target='_blank'>`,'</a>')"></p>
+          <input
+            type="button"
+            role="validation"
+            :value="translate('Feedback_SendFeedback')"
+          />
+          <input
+            type="button"
+            role="cancel"
+            :value="translate('General_Cancel')"
+          />
+        </div>
+      </MatomoDialog>
+      <MatomoDialog v-model="feedbackDone">
+        <div
+          class="ui-confirm ratefeatureDialog"
+        >
+          <h2>{{ translate(`Feedback_ThankYou`) }}</h2>
+          <p v-html="translate('Feedback_ThankYourForFeedback',
+        `<i class='icon-heart red-text'></i>`)">
+          </p>
+          <input
+            type="button"
+            role="cancel"
+            :value="translate('General_Close')"
+          />
+        </div>
+      </MatomoDialog>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue';
+import {
+  MatomoDialog, AjaxHelper, setCookie, getCookie, translate,
+} from 'CoreHome';
+
+const { $ } = window;
+
+const cookieName = 'feedback-question';
+export default defineComponent({
+
+  props: {
+    showQuestionBanner: String,
+  },
+  components: {
+    MatomoDialog,
+  },
+  computed: {
+    isHidden() {
+      if (this.showQuestionBanner === '0') {
+        return true;
+      }
+      return !!this.hide;
+    },
+  },
+  data() {
+    return {
+      questionText: '',
+      question: 0,
+      hide: null,
+      feedbackDone: false,
+      expanded: false,
+      showFeedbackForm: false,
+      feedbackMessage: null,
+      errorMessage: null,
+    };
+  },
+  watch: {
+    showFeedbackForm(val) {
+      // eslint-disable-next-line no-underscore-dangle
+      this.questionText = translate(`Feedback_Question${this.question}`);
+      if (val) {
+        setInterval(() => {
+          $('#message').focus();
+        }, 500);
+      }
+    },
+  },
+  created() {
+    if (this.showQuestionBanner !== '0') {
+      // eslint-disable-next-line no-restricted-globals
+      if (location.href.includes('forceFeedbackTest')) {
+        this.question = 0;
+        setCookie(cookieName, 0, 7);
+      } else {
+        this.initQuestion();
+      }
+    }
+  },
+  methods: {
+    initQuestion() {
+      if (getCookie(cookieName)) {
+        // eslint-disable-next-line radix
+        this.question = parseInt(getCookie(cookieName));
+        const nextQuestion = (this.question + 1 > 4) ? 0 : this.question + 1;
+        setCookie(cookieName, nextQuestion, 7);
+      } else {
+        setCookie(cookieName, this.getRandomIntBetween(0, 4), 7);
+      }
```

Comment:
Actually the first question shown is still always number 0. Guess this is as initially there is no cookie, and the cookie set in that case is then used for the next page load. Guess this could be fixed and simplified with this:
```suggestion
      if (!getCookie(cookieName)) {
        this.question = this.getRandomIntBetween(0, 4);
      } else {
        // eslint-disable-next-line radix
        this.question = parseInt(getCookie(cookieName));
      }
      
      const nextQuestion = (this.question + 1) % 4;
      setCookie(cookieName, nextQuestion, 7);
```